lib/config/types.ts

```diff
@@ -161,15 +160,21 @@ export type RenovateRepository =
       repository: string;
       secrets?: Record<string, string>;
     };
-
-export interface CustomManager {
-  fileMatch: string[];
-  matchStrings: string[];
-  matchStringsStrategy?: string;
+export interface RegexManagerTemplates {
   depNameTemplate?: string;
-  datasourceTemplate?: string;
   packageNameTemplate?: string;
+  datasourceTemplate?: string;
   versioningTemplate?: string;
+  depTypeTemplate?: string;
+  currentValueTemplate?: string;
+  currentDigestTemplate?: string;
+  extractVersionTemplate?: string;
+  registryUrlTemplate?: string;
+}
+export interface CustomManager extends RegexManagerTemplates {
```

Comment:
we should rename this to `RegexManager`

maybe another thing to extract to separate PR. 🤔

Reply:
Yes, will rename in another PR once we have correctly implemented narrowing as well.

Reply:
Renaming should be done separately from any functional change